sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionUtilityToolFilter.java

```diff
@@ -0,0 +1,171 @@
+	protected static final String PROPERTY_TEST_URL = "test.url";
+	private static final String AUTH_GROUP = "its-canvas-sections";
+	private static final String TEST_USER = "testUser";
+	protected static Properties canvasProperties = null;
+	private boolean isTestUrlEnabled=false;
+
+	@Override
+	public void init(FilterConfig filterConfig) throws ServletException {
+		M_log.debug("Filter Init(): Called");
+		getCanvasCredentials();
+		
+	}
+
+	@Override
+	public void doFilter(ServletRequest request, ServletResponse response,
+			FilterChain chain) throws IOException, ServletException {
+		M_log.debug("doFilter: Called");
+		HttpServletRequest useRequest = (HttpServletRequest) request;
+		HttpServletResponse useResponse=(HttpServletResponse)response;
+		if(!checkForAuthorization(useRequest)) {
+		      useResponse.sendError(403);
+			return;
+		}
+		chain.doFilter(useRequest, response);
+	}
+
+	@Override
+	public void destroy() {
+		M_log.debug("detroy: Called");
+	}
+	protected void getCanvasCredentials() {
+		M_log.debug("getCanvasCredentials(): called");
+		String propertiesFilePath = System.getProperty(SYSTEM_PROPERTY_FILE_PATH);
+		if (!isEmpty(propertiesFilePath)) {
+		canvasProperties=PropertiesUtilities.getPropertiesObjectFromURL(propertiesFilePath);
+		}else {
+			M_log.error("File path for (sectionsToolProps.properties) is not provided");
+		}
+		
+		
+	}
+	 private boolean isEmpty(String value) {
+		return (value == null) || (value.trim().equals(""));
+	}
+	
+	/*
+	 * User is authenicated using cosign  authorized using Ldap. For local development we are enabling
+	 * "testUser" parameter. "testUser" also go through the Ldap authorization process. 
+	 * we have test.url configured in the properties file in order to disable usage of "testUser" parameter in PROD.
+	 * "/assets" folder is where all the js/css resources. I am not making those folder go through authorization process for
+	 * 2 reasons 1) If the request of /index.html page fails we return the error page and that needs .css file for doing styling.
+	 * 2) if /index.html is 200 ok, the next request that follows are /assets. so it kind of redundant to go through ldap check.
+	 */
+	private boolean checkForAuthorization(HttpServletRequest request) {
+		M_log.debug("checkLdapForAuthorization(): called");
+     	String remoteUser = request.getRemoteUser();
+		String servletPath = request.getServletPath();
+		String testUser = request.getParameter(TEST_USER);
+		boolean isAuthorized = false;
+		
+		if(canvasProperties!=null) {
+			isTestUrlEnabled = Boolean.parseBoolean(canvasProperties.getProperty(SectionUtilityToolFilter.PROPERTY_TEST_URL));
+		}else {
+			M_log.warn("Failed to load system property test.url from sectionsToolProps.properties for SectionsTool");
+		}
+		
+		if(servletPath.startsWith(ASSETS_PATH)) {
+			return true;
+		}
+		String testUserInSession = (String)request.getSession().getAttribute(TEST_USER);
+		
+	    if ( isTestUrlEnabled && testUser != null ) { 
```

Comment:
It would be clear to have 1 call to ldapAuthorizationVerification and to use the if block just to set the user name what will be verified.